README.md

```diff
@@ -1,77 +1,65 @@
-- Report progress to the user throughout the download
-- Add the ability to cancel the download
-
-
-We are looking for you to demonstrate:
-- Analysing requirements
-- Solving problems
-- Using language features appropriately
-- Structuring code
-- Verifying your solution works correctly
-
-If you feel that modifying the skeleton project would create a better solution to these requirements, we encourage you to do so. We currently support submissions in either .NET (C#) or Java and have included skeleton projects for both.
-
-### .NET 
-
-There is already a ```IWebSystemCalls.cs``` and corresponding implementation which allows you to get the HTTP headers for a URL, download the whole content, or download partial content. All these calls return an ```HttpResponseMessage``` object which contains properties for headers and the content.
-
-As in the example here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Ranges, the HTTP header "Accept Ranges" will be set to "Bytes" if the CDN supports partial content.
+We would like you to update the skeleton project in this repository to provide a way for clinicians to download our software given multiple network conditions. Details can be found in the requirements section below. 
 
-An NUnit test project is included if you would like to add unit tests - feel free to swap out NUnit for any other test framework you may prefer. There is also a Program Main function which has a real Accurx URL (which **DOES** support partial content) and an example of a file path so that the code can be tested for real.
+### Requirements
 
-### Java
+1. The installer downloads in the following scenarios:
+   - When internet disconnections occur (we use 2 minutes as a disconnection time benchmark for this)
+   - Partial downloading. So that the download doesn't need to start from scratch every time if the CDN supports this
+   - Downloading the file in one go, if the CDN does not support partial downloading
+2. The system can recover from failures and not exit until the file has been successfully downloaded
+3. The file is deleted if the integrity check fails after downloading. You can use the Content-MD5 for this: https://www.oreilly.com/library/view/http-the-definitive/1565925092/re17.html
+4. Progress is reported to the user throughout the download
+5. The user can cancel the download
 
-There is already a ```WebSystemCalls.java``` and corresponding implementation which allows you to get the HTTP headers for a URL, download the whole content or download the partial content. All these calls return an ```HttpResponse``` object, which contains sub properties for headers and the content.
+### What does a successful submission look like?
+A high-quality submission will have the following:
+- Meet the requirements above
+- Appropriate use of language features to solve the problem in a neat and simple way
+- Well structured code
```

Comment:
What does well-structured code look like? Can we be more specific about the things we're looking for here?

Reply:
I think we're looking for code that is easy to read and reason about. Typically using software design principles like SOLID.

Reply:
Thanks Jaf! I've made an update based on this - really helpful!